### DISCUSSION TITLE
Row selection issue in DataGrid

### DISCUSSION THREAD

**Discussion by zg4455** [AvaloniaUI/Avalonia#16196] [Q&A]
### Describe the bug

https://github.com/AvaloniaUI/Avalonia/assets/25166463/682a481c-a8ed-4200-8fa3-46f6c61fc559

### To Reproduce

1. Select the first row of the DataGrid
2. Click the button to modify the first row
3. After successful modification, the second row will be automatically selected

### Expected behavior

I don't want to change the selected row. Is there any way

### Avalonia version

11.1.0

### OS

Windows

### Additional context

Bind datagrid to ObservableCollection and modify the line using the following code
Items[1]. ID = Guid.NewGuid();
Items[1] = Items[1];

**Comment by timunie** [comment]
if you replace the item, you need to set the Selection back after the item was replaced. Also please tell us exactly why you think this is a bug?

**Comment by zg4455** [comment]
How to modify row data without replacing the Item?

**Reply by timunie** [reply]
the items need to implement INotifyPropertyChanged (ViewModelBase has it for example)
